### PR TITLE
Stonesoup UI Test: Update Basic happy path & Create Advanced happy path tests

### PR DESCRIPTION
**Pull request by dheerajodha**
## Fixes 

https://issues.redhat.com/browse/HACBS-1665
https://issues.redhat.com/browse/HACBS-1666

## Type of change

- [x] Other (please describe): Test

## Screen shots / Gifs for design review 

Test video for [Basic happy path](https://drive.google.com/file/d/1Lj_4-OTBI9M8CqDayickIMlAYSU0l85B/view) and for [Advanced happy path](https://drive.google.com/file/d/1XljkcFwZrM3xxCzRd04WIWeELVyGXrIp/view?usp=share_link).

## Browser conformance: 

- [x] Chrome

### CODE REVIEW

**Comment by dheerajodha** [comment]
2 TODO tasks for this PR:

1. The test isn't able to click on the "Create" button after it ticks the "Send pull request" (reproducible locally as well):
image

2. The test needs to have a function that deletes a folder (`.tekton/` in our case) using GitHub API, to be used in the cleanup task.

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        // cy.get(pipelinerunsTabPO.statusPO).invoke('text').then(text => {
+        //     if (text.includes('Succeeded')) {
+        //         TaskRunsTab.goToTaskRunsTab().assertTaskNamesAndTaskRunStatus(isAdvancedFlowActive);
+        //     }
+        //     else if (text.includes('Failed')) {
+        //         LogsTab.goToLogsTab();
+        //         cy.screenshot('capture-logs-on-pipelinerun-failure', {capture: 'fullPage'});
+        //     }
+        // });
```
Comment (jrichter1) [inline integration-tests/support/pages/tabs/PipelinerunsTabPage.ts:31]: remove if not necessary please

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        //         cy.get(pipelinerunsTabPO.statusPO, { timeout: 720000 }).should('not.have.text', 'Running');
+        //     }
+
+        //     if (text.includes('Failed')) {
+        //         LogsTab.goToLogsTab();
+        //         cy.screenshot('capture-logs-on-pipelinerun-failure', {capture: 'fullPage'});
+        //     }
+        // })
+
+        //cy.wait(25000);
```
Comment (jrichter1) [inline integration-tests/support/pages/tabs/PipelinerunsTabPage.ts:56]: same as above

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            else if (text.includes('Failed')) {
+                LogsTab.goToLogsTab();
+                cy.screenshot('capture-logs-on-pipelinerun-failure', {capture: 'fullPage'});
+            }
+        });
+    }
+}
+
+export class TaskRunsTab {
+    taskNameList: string[] = ['init', 'git-clone', 'configure-build', 'buildah', 'summary',
```
Comment (jrichter1) [inline integration-tests/support/pages/tabs/PipelinerunsTabPage.ts:76]: can this be pulled from somewhere dynamically?

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                                    'sbom-json-check', 'sanity-inspect-image', 'clair-scan',
+                                    'sanity-label-check', 'sanity-label-check'];
+
+    static goToTaskRunsTab() {
+        cy.get(pipelinerunsTabPO.clickTaskRunsTab).click();
+        return new TaskRunsTab();
+    }
+
+    assertTaskNamesAndTaskRunStatus(isAdvancedFlowActive: boolean = false) {
+        for (let i = 0; i < ((isAdvancedFlowActive)? this.taskNameList.length : 5); i++) {
```
Comment (jrichter1) [inline integration-tests/support/pages/tabs/PipelinerunsTabPage.ts:87]: I'm not a fan of the 5 there. Maybe separate the basic and advanced tasks, and only merge them for the advanced scenario?

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            cy.get(`[data-test="${i}-0"] > td`).eq(3).within(() => {
+                cy.get(pipelinerunsTabPO.taskRunStatus).invoke('text').then(status => {
+                    if (status.includes('Failed')) {
+                        cy.screenshot('taskrun-failed-even-though-pipelinerun-succeeded', {capture: 'fullPage'});
+                        return;
+                    }
+                    else if (status.includes('Pending')) {
+                        cy.screenshot('taskrun-pending-even-though-pipelinerun-succeeded', {capture: 'fullPage'});
+                        return;
+                    }
```
Comment (jrichter1) [inline integration-tests/support/pages/tabs/PipelinerunsTabPage.ts:101]: shouldn't the test just fail if the pipeline doesn't succeed? in which case the screenshot would be taken automatically

File: integration-tests/tests/advanced-happy-path.spec.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+import { Common } from '../utils/Common';
+
+describe('Create Components using the UI', () => {
+  const applicationName = Common.generateAppName();
+  const applicationDetailPage = new ApplicationDetailPage();
+  const pipelinerunsTab = new PipelinerunsTabPage();
+  const latestCommitsTabPage = new LatestCommitsTabPage();
+  const pipelineName = 'demo-pipeline';
+  const publicRepos = [
+    'https://github.com/dheerajodha/devfile-sample-go-basic',
```
Comment (jrichter1) [inline integration-tests/tests/advanced-happy-path.spec.ts:15]: don't forget to update the repo when done

File: integration-tests/tests/advanced-happy-path.spec.ts [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+      }
+    });
+
+    it("Verify that the component deployment reflects latest changes", () => {
+      Applications.goToComponentsTab();
+
+      applicationDetailPage.expandDetails(componentName);
+
+      cy.get(applicationDetailPagePO.route.replace('{0}', componentName), { timeout: 35000 }).invoke('text').then(route => {
+        cy.wait(240000);
```
Comment (jrichter1) [inline integration-tests/tests/advanced-happy-path.spec.ts:158]: we don't like long unconditional waits here

File: integration-tests/tests/advanced-happy-path.spec.ts [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+        componentInfo.get('goFileBase64Updated'),
+        componentInfo.get('goFileSHAOriginal')
+      );
+    });
+
+    it("Check the new Pipeline run", () => {
+      Applications.goToActivityTab();
+      Applications.goToPipelinerunsTab();
+
+      cy.wait(10000);
```
Comment (jrichter1) [inline integration-tests/tests/advanced-happy-path.spec.ts:130]: does it have to wait 10 seconds?

File: integration-tests/tests/advanced-happy-path.spec.ts [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+  });
+
+  describe('Check Component Deployment', () => {
+    it("Verify the status code and response body of the deployment URL of each component", () => {
+      Applications.goToComponentsTab();
+
+      applicationDetailPage.expandDetails(componentName);
+
+      cy.get(applicationDetailPagePO.route.replace('{0}', componentName), { timeout: 35000 }).invoke('text').then(route => {
+        cy.wait(20000);
```
Comment (jrichter1) [inline integration-tests/tests/advanced-happy-path.spec.ts:102]: what are we waiting for here and why 20 seconds?

File: integration-tests/tests/basic-happy-path.spec.ts [hunk 10]
```diff
@@ -34,44 +40,91 @@ describe('Create Components using the UI', { tags: ['@PR-check', '@publicRepo']
...
 
+  describe('Check Component Deployment', () => {
+    it("Verify the status code and response body of the deployment URL of each component", () => {
+      Applications.goToComponentsTab();
+
+      for (let componentName of componentNames) {
+        applicationDetailPage.expandDetails(componentName);
+
+        cy.get(applicationDetailPagePO.route.replace('{0}', componentName), { timeout: 35000 }).invoke('text').then(route => {
+          cy.wait(30000);
```
Comment (jrichter1) [inline integration-tests/tests/basic-happy-path.spec.ts:113]: we don't like long unconditional waits here

File: integration-tests/tests/basic-happy-path.spec.ts [hunk 11]
```diff
@@ -34,44 +40,91 @@ describe('Create Components using the UI', { tags: ['@PR-check', '@publicRepo']
...
+            expect(response.body).to.include(deploymentBody.get(componentName));
+          })
+        });
+      }
+
+      Applications.checkComponentStatus(componentNames[0], 'Build Succeeded');
+    })
+  });
+
   describe.skip('Explore Integration Tests Tab', { tags: '@IntegrationTestsTab' }, () => {
```
Comment (jrichter1) [inline integration-tests/tests/basic-happy-path.spec.ts:128]: can you remove this from the suite for the MVP?

File: integration-tests/utils/Applications.ts [hunk 12]
```diff
@@ -40,17 +43,39 @@ export class Applications {
...
   static createdComponentExists(componentName: string, applicationName: string) {
     this.goToComponentsTab();
 
     Common.verifyPageTitle(applicationName);
     Common.waitForLoad();
     this.getComponentListItem(componentName).should('exist');
+
+    // if (strictChecking) {
+    //   cy.get(componentsTabPO.componentListItem.replace('{0}', componentName)).contains(/.*Build Running.*/, { timeout: 5000 });
+    // }
```
Comment (jrichter1) [inline integration-tests/utils/Applications.ts:61]: remove or rework, depending if it's needed or not

**Review by jrichter1 — CHANGES_REQUESTED** [review]

File: integration-tests/support/pageObjects/createApplication-po.ts [hunk 13]
```diff
@@ -58,6 +60,7 @@ export const applicationDetailPagePO = {
   detailsArrow: '[aria-label="Details"]',
   cpuRamLabel: 'CPU / Memory',
   replicaLabel: 'Instances',
+  route: '[data-test-id="{0}-route"]',
```
[the remark this answers is not in the document]
Comment (dheerajodha) [reply integration-tests/support/pageObjects/createApplication-po.ts]: very clean, thanks for suggesting! done

File: integration-tests/support/pages/tabs/LatestCommitsTabPage.ts [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    this.owner = publicGitRepo.split("/")[3];
+    this.repoName = publicGitRepo.split("/")[4];
+
+    cy.exec(`curl -s \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer ${this.token}"\
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/search/issues?q=${componentName}`).then((searchIssueResponse) => {
+        const obj = JSON.parse(searchIssueResponse.stdout);
+        const pullNumber = obj["items"][0]["number"];
```
[the remark this answers is not in the document]
Comment (dheerajodha) [reply integration-tests/support/pages/tabs/LatestCommitsTabPage.ts]: thanks for helping! done

File: integration-tests/support/pages/tabs/LatestCommitsTabPage.ts [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+              sha: "",
+              title: "",
+              user: "",
+              message: ""
+            };
+
+            commitObject.sha = commitsArray[i]["sha"];
+            commitObject.title = `#${pullNumber} Appstudio update ${componentName}`;
+            commitObject.user = "appstudio-staging-ci__bot";
+            commitObject.message = "";
```
[the remark this answers is not in the document]
Comment (dheerajodha) [reply integration-tests/support/pages/tabs/LatestCommitsTabPage.ts]: done, thanks

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts (hunk 1 above)
Comment (dheerajodha) [reply integration-tests/support/pages/tabs/PipelinerunsTabPage.ts]: sorry for this, done

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts (hunk 2 above)
Comment (dheerajodha) [reply integration-tests/support/pages/tabs/PipelinerunsTabPage.ts]: sorry for this too, done

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts (hunk 4 above)
Comment (dheerajodha) [reply integration-tests/support/pages/tabs/PipelinerunsTabPage.ts]: done, thanks

File: integration-tests/support/pages/tabs/PipelinerunsTabPage.ts (hunk 5 above)
Comment (dheerajodha) [reply integration-tests/support/pages/tabs/PipelinerunsTabPage.ts]: makes sense. done, thanks

File: integration-tests/tests/advanced-happy-path.spec.ts (hunk 7 above)
Comment (dheerajodha) [reply integration-tests/tests/advanced-happy-path.spec.ts]: done, thanks

File: integration-tests/tests/advanced-happy-path.spec.ts (hunk 8 above)
Comment (dheerajodha) [reply integration-tests/tests/advanced-happy-path.spec.ts]: done, thanks

File: integration-tests/tests/advanced-happy-path.spec.ts (hunk 9 above)
Comment (dheerajodha) [reply integration-tests/tests/advanced-happy-path.spec.ts]: done, thanks

File: integration-tests/tests/basic-happy-path.spec.ts (hunk 10 above)
Comment (dheerajodha) [reply integration-tests/tests/basic-happy-path.spec.ts]: done, thanks

File: integration-tests/tests/basic-happy-path.spec.ts (hunk 11 above)
Comment (dheerajodha) [reply integration-tests/tests/basic-happy-path.spec.ts]: done, thanks

File: integration-tests/utils/Applications.ts (hunk 12 above)
Comment (dheerajodha) [reply integration-tests/utils/Applications.ts]: removed, and created a new function `checkComponentStatus()` with similar functionality

File: integration-tests/tests/advanced-happy-path.spec.ts [hunk 16]
```diff
@@ -0,0 +1,168 @@
+import { applicationDetailPagePO, ComponentsPagePO } from '../support/pageObjects/createApplication-po';
+import { ApplicationDetailPage } from '../support/pages/ApplicationDetailPage';
+import { ComponentPage } from '../support/pages/ComponentsPage';
+import { LatestCommitsTabPage } from '../support/pages/tabs/LatestCommitsTabPage';
+import { PipelinerunsTabPage } from '../support/pages/tabs/PipelinerunsTabPage';
+import { Applications } from '../utils/Applications';
+import { Common } from '../utils/Common';
+
+describe.skip('Create Components using the UI', () => {
```
Comment (subhashkhileri) [inline integration-tests/tests/advanced-happy-path.spec.ts:9]: Please comment the entire spec file as the skipping test is causing issue on uploading the result on Report Portal which leads to launch timeout.

Comment (dheerajodha) [reply integration-tests/tests/advanced-happy-path.spec.ts]: Yes, just did that, thanks!

**Pull request merged**